lib.js

```diff
@@ -93,3 +93,29 @@ export const generateTestsuiteResult = (suiteResult) => {
 
   return result.join(EOL);
 };
+
+export const generateTestsuiteLogs = (suiteResult) => {
+  let result = [];
+  const testscases = suiteResult.testcase;
+
+  if (testscases) {
+    let errLogs = [...testscases.map(testcase => formatLogLine(testcase['system-err']))].filter(a => a !== '');
+    if (errLogs.length) {
+      result.push(EOL, '   > Error Log output:', EOL, EOL, '\t', ...errLogs);
+    }
+
+    let outLogs = [...testscases.map(testcase => formatLogLine(testcase['system-out']))].filter(a => a !== '');
+    if (outLogs.length) {
+      result.push(EOL, '   > Standard Log output:', EOL, EOL, '\t', ...outLogs);
+    }
+  }
+
+  return result.join('');
```

Comment:
```suggestion
  const errLogs = testscases.map(testcase => formatLogLine(testcase['system-err'])).filter(a => a);
  if (errLogs.length) {
    result.push(`   ${logSymbols.info} Error Log output:`, '\t', ...errLogs);
   }

   const outLogs = testscases.map(testcase => formatLogLine(testcase['system-out'])).filter(a => a);
   if (outLogs.length) {
     result.push(`   ${logSymbols.info} Standard Log output:`, '\t', ...outLogs);
   }
 }

  return result.join(EOL);
```

This can be better shorted like above. testscases is already an Array and hence the map and spread to create a new Array won't be required. Also, just filter would remove empty strings in Node. Hence extra check for empty is not needed.

Reply:
Updated 👍